producer.go

```diff
@@ -243,7 +245,7 @@ func (p *Producer) flush(records []*kinesis.PutRecordsRequestEntry, reason strin
 
 	for {
 		p.Logger.Info("flushing records", LogValue{"reason", reason}, LogValue{"records", len(records)})
-		out, err := p.Client.PutRecords(&kinesis.PutRecordsInput{
+		out, err := p.Client.PutRecords(context.TODO(), &kinesis.PutRecordsInput{
```

Comment:
We should probably allow a `context.Context` to be passed into the `Start` method and propagate it here, but that would imply that we're handling a cancelled context in the dispatch loop. Implementing that handling seems beyond the scope of this migration, so this is just explicitly opting out of it for the time being.